MICROSOFT ENTERS THE MOVIES BUSINESS

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plot
import sqlite3
%matplotlib inline

First, we need to look at the data we are working with before proceeding.

We will start with tmdb.movies.csv

In [3]:
tmdb_file_location = '../tmdb.movies.csv'

tmdb_movies_df = pd.read_csv(tmdb_file_location)
tmdb_movies_df.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [4]:
tmdb_movies_df.columns

Index(['Unnamed: 0', 'genre_ids', 'id', 'original_language', 'original_title',
       'popularity', 'release_date', 'title', 'vote_average', 'vote_count'],
      dtype='object')

In [5]:
tmdb_movies_df.shape

(26517, 10)

tmdb.movies.csv has the following 10 columns 'Unnamed: 0', 'genre_ids', 'id', 'original_language', 'original_title', 'popularity', 'release_date', 'title', 'vote_average', 'vote_count''

It has 10 columns and 26,517 rows

Looks like this table summarizes the movies popularity with the fans.

Next, lets see what data is in tn.movie_budgets.

In [6]:
tn_movie_budget_path = '../tn.movie_budgets.csv'

movie_budgets_df = pd.read_csv(tn_movie_budget_path)

movie_budgets_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [9]:
movie_budgets_df.shape

(5782, 6)

In [8]:
movie_budgets_df.columns

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')

tn.movie_budgets.csv has the following 6 columns 'id', 'release_date', 'movie', 'production_budget', 'domestic_gross', 'worldwide_gross'.

It has 5782 rows and 6 columns

Looks like this table sumarizes how successful movies are from a monetary perspective.

Next we will connect to the database and look at what is in it

From the ERD, the table 'movie_basics' looks like a good place to start

In [29]:


conn = sqlite3.connect('../im.db')

cur = conn.cursor()



In [30]:
cur.execute(""" SELECT name FROM sqlite_master WHERE type = 'table';""")

tables = cur.fetchall()

print(tables)

[('movie_basics',), ('directors',), ('known_for',), ('movie_akas',), ('movie_ratings',), ('persons',), ('principals',), ('writers',)]


The imdb database has the following tables 'movie_basics', 'directors', 'known_for', 'movie_akas','movie_ratings', 'persons', 'principals', 'writers'

Next we will load this data into a dataframe for easier inspection and review.

In [45]:
movie_basics = pd.read_sql("""
                      SELECT *
                      FROM movie_basics;""" ,conn )
movie_basics.head()

ProgrammingError: Cannot operate on a closed database.

In [42]:
directors = pd.read_sql("""
                      SELECT *
                      FROM directors;""" ,conn )
directors.head()

movie_id  person_id
0  tt0285252  nm0899854
1  tt0462036  nm1940585
2  tt0835418  nm0151540
3  tt0835418  nm0151540
4  tt0878654  nm0089502

In [43]:
persons = pd.read_sql("""
                      SELECT *
                      FROM persons;""" ,conn )
persons.head()

person_id       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  
0         miscellaneous,production_manager,producer  
1        composer,music_department,sound_department  
2                        miscellaneous,actor,writer  
3  camera_department,cinematographer,art_department  
4  production_designer,art_department,set_decorator